### ISSUE TITLE
Make --replication-lag-query obsolete?

### ISSUE THREAD

**Issue by shlomi-noach** [github/gh-ost#346]
Driven by a question by @krzysztof-ksiazek, and see [relevant documentation](https://github.com/github/gh-ost/blob/d13049b41e9da9798b83958b51f8ecf6620bb100/doc/subsecond-lag.md)

At this time there are two independent replication lag measurements: 

- Internal `_ghc` heartbeat injection & reads
- `--replication-lag-query` reads, typically reading value of `pt-heartbeat`-injected heartbeat.

`gh-ost` injects a heartbeat onto the `_ghc` (_changelog_) table every `--heartbeat-interval-millis` (default at this time: `500`, ie `0.5 seconds`). It does so because we make no assumption on the master having or not having a `pt-heartbeat` or other heartbeat interval of its own.

The _changelog_ table is there anyway, and is being used throughout the migration for a variety of reasons. It is therefore a good idea to utilize it for heartbeats.

On a "normal" migration (i.e. not "test on replica" or "migrate on replica"), `gh-ost` writes the heartbeat on the master, then reads the timestamp on the replica it connects to. When running a "migrate on master" migration, we read the heartbeat from the very master. This is the "internal" heartbeat control, and it's always there.

In addition to the above, `gh-ost` also allows for a list of `--throttle-control-replicas`. At this time, `gh-ost` measures lag on the throttle control replicas via either:

- `--replication-lag-query`, when given.
- and when not given, `Seconds_behind_master` column of `show slave status` result.

### Why two methods?

Originally internal heartbeat was read by tailing the binary logs of the migration replica. Thus, heartbeat was injected on master, and collected via binlog event on replica.

We could not rely on all replicas having binary logs, therefore could not use the same collection method, and thus the reason for `--replication-lag-query` or `show slave status` as means of collecting lag.

### But then

`gh-ost`'s lag collection logic changed to reading directly from the _changelog_ table. Reason was a catch-22 problem: if the replica was lagging, we would throttle, meaning we would suspend reading the binary logs. But then, what would make us determine if replication caught up and lag was eliminated?

So, at this time `gh-ost` reads the _changelog_ table, and there is no reason why it wouldn't do the same on the `--throttle-control-replicas`.

### TL;DR

Suggestion: make `--replication-lag-query` obsolete; no need to rely on external `pt-heartbeat` mechanism of any sort. Rely on _changelog_ table injection at all places.

**Comment by shlomi-noach** [comment]
closed by #347 , released in https://github.com/github/gh-ost/releases/tag/v1.0.34
